paprika/restraints/plumed.py

```diff
@@ -0,0 +1,348 @@
+import logging
+import os
+
+import numpy as np
+from paprika.dummy import extract_dummy_atoms
+from paprika.restraints.utils import get_bias_potential_type, parse_window
+from paprika.utils import get_key, return_parmed_structure
+from parmed.structure import Structure as ParmedStructureClass
+
+logger = logging.getLogger(__name__)
+
+_PI_ = np.pi
+
+
+class Plumed:
+    """
+    This class is converts restraints generated in pAPRika DAT_restraints into Plumed restraints.
+
+    Example:
+    -------
+        >>> plumed = Plumed()
+        >>> plumed.file_name = 'plumed.dat'
+        >>> plumed.path = './windows'
+        >>> plumed.window_list = window_list
+        >>> plumed.restraint_list = restraint_list
+        >>> plumed.dump_to_file()
+
+    plumed.dat:
+        UNITS LENGTH=A ENERGY=kcal/mol TIME=ns
+        # Collective variables
+        c1 : DISTANCE ATOMS=175,150 NOPBC
+        c2 : ANGLE    ATOMS=176,175,150 NOPBC
+        c3 : ANGLE    ATOMS=175,150,165 NOPBC
+        # Bias potential
+        RESTRAINT   ARG=c7  AT=  6.000 KAPPA=  10.00
+        RESTRAINT   ARG=c8  AT=  3.142 KAPPA= 200.00
+        RESTRAINT   ARG=c9  AT=  3.142 KAPPA= 200.00
+    """
+
+    @property
+    def file_name(self):
+        """
+        str: The Plumed file name where the restraints will be written to.
+        """
+        return self._file_name
+
+    @file_name.setter
+    def file_name(self, value: str):
+        self._file_name = value
+
+    @property
+    def window_list(self):
+        """
+        list: The list of APR windows
+        """
+        return self._window_list
+
+    @window_list.setter
+    def window_list(self, value: list):
+        self._window_list = value
+
+    @property
+    def restraint_list(self):
+        """
+        list: The list of restraints to be converted to Plumed.
+        """
+        return self._restraint_list
+
+    @restraint_list.setter
+    def restraint_list(self, value: list):
+        self._restraint_list = value
+
+    @property
+    def path(self):
+        """
+        os.PathLike: The parent directory that contains the simulation windows.
+        """
+        return self._path
+
+    @path.setter
+    def path(self, value: str):
+        self._path = value
+
+    @property
+    def uses_legacy_k(self):
+        """
+        bool: Option to specify whether the force constant parsed into DAT_restraint()
+        is halved or not (i.e. value is already multiplied by a factor of 1/2).
```

Comment:
To be clear, `uses_legacy_k` means the value is half of what you'd expect? Might be clear to add the note `uses_legacy_k=True` means Amber-style, and `uses_legacy_k=False` means CHARMM/NAMD-style.

Reply:
Yes, and a very good point. I'll make this point more apparent in the docstring.